extensions/amp-cxense-player/0.1/amp-cxense-player.js

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {CSS} from '../../../build/amp-cxense-player-0.1.css';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {appendParamStringToUrl, parseUrl} from '../../../src/url';
+import {dashToCamelCase} from '../../../src/string';
+import {setStyles} from '../../../src/style';
+const extend = require('extend');
+
+const cxDefaults = {
+  apiHost: 'https://api.widget.cx',
+  embedHost: 'https://embed.widget.cx',
+  distEmbedApp: '/app/player/m4/dist/',
+  debugEmbedApp: '/app/player/m4/debug/',
+  attrs: {
+        // can't access the window.top.location.href from the iframe.
+    share: {
+      enable: false,
+    },
+  },
+};
+
+class AmpCxense extends AMP.BaseElement {
+
+    /** @override */
+    preconnectCallback(onLayout) {
+      this.preconnect.url(cxDefaults.apiHost, onLayout);
+      this.preconnect.prefetch(
+            cxDefaults.embedHost + cxDefaults.distEmbedApp, 'iframe'
```

Comment:
This should be the same as the actual iframe src. Can you just call that same method?

@mkhatib Is `iframe` supported here? If yes (and it would make sense), we might do this wrong elsewhere.